src/codegen/abi.js

```diff
@@ -502,11 +489,7 @@ module.exports = class AbiCodeGenerator {
       let params = inputs.map((input, index) =>
         tsCodegen.param(
           input.get('name'),
-          input.get('type') === 'tuple'
-            ? this._tupleTypeName(input, index, tupleInputParentType, this.abi.name)
-            : util.isTupleArrayType(input.get('type'))
-              ? `Array<${this._tupleTypeName(input, index, tupleInputParentType)}>`
-              : typesCodegen.ascTypeForEthereum(input.get('type')),
+          this._getTupleParamType(input, index, tupleResultParentType),
```

Comment:
`tupleResultParentType` needs to be `tupleInputParentType` (see the old code just above this addition). `tupleResultParentType` is only used for return values, not input arguments.